app/Models/Article.php

```diff
@@ -369,4 +370,33 @@ protected function transformMappingInternal()
             ],
         ];
     }
+
+    public function related($id)
+    {
+        $modelName = 'articles';
+
+        $blockableTypes = [
+            'highlights',
+            'educatorResources',
+            'articles',
+            'videos',
+            'researchGuides',
+            'genericPages',
+            'events',
+            'exhibitions',
+        ];
+
+        $relatedBlockItems = Block::whereIn('blockable_type', $blockableTypes)
+            ->where('content', 'LIKE', '%"browsers"%')
+            ->where('content', 'LIKE', '%' . $modelName . '%')
+            ->where('content', 'LIKE', '%' . $id . '%')
+            ->get();
+
+        $relatedItems = [];
+        foreach ($relatedBlockItems as $relatedBlockItem) {
+            $relatedItems[] = $relatedBlockItem->blockable;
+        }
+
+        return $relatedItems;
+    }
```

Comment:
Rather than repeating this method in all four classes, create a new [trait](https://alexwebdevelop.com/php-traits/) and use it in these three classes.

From what I can tell the only thing that's different between these methods is the `$modelName = 'articles';` assignment. This can be determined instead of set statically:

```
class_basename(__CLASS__);
```

That will give you the `Article` portion of `App\Models\Article`. If you then want it lowercase and plural, add `use Illuminate\Support\Str;` at the top of the class then:

```
Str::plural(Str::lower(class_basename(__CLASS__)));
```